### PR TITLE
feat: add failoverDelay cluster parameter

### LINKED ISSUE #507: Adding a failoverDelay parameter to wait sometime before triggering a failover
Following this discussion https://github.com/cloudnative-pg/cloudnative-pg/discussions/466 I worked on adding a feature to delay the triggering of a failover when the primary is considered unhealthy.

### PR DESCRIPTION
**Pull request by armru**
Closes #507

### CODE REVIEW

File: config/crd/bases/postgresql.cnpg.io_clusters.yaml [hunk 1]
```diff
@@ -2104,6 +2104,12 @@ spec:
                   - name
                   type: object
                 type: array
+              failoverDelay:
+                default: 0
+                description: The time in seconds a primary PostgreSQL instance should
+                  be deemed unhealthy before triggering a failover
```
Comment (gbartolini) [inline config/crd/bases/postgresql.cnpg.io_clusters.yaml:2110]: I would rephrase this a little bit. Maybe something like: _The amount of time (in seconds) to wait before triggering a failover after the primary PostgreSQL instance in the cluster was detected to be unhealthy_

**Review by mnencia — APPROVED** [review]

**Review by litaocdl — APPROVED** [review]
